src/components/MarkdownProvider/components/CodeExample/index.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useMemo, useEffect } from 'react';
+import styled, { css, DefaultTheme } from 'styled-components';
+import { ThemeProvider, DEFAULT_THEME, getColor } from '@zendeskgarden/react-theming';
+import { Tooltip } from '@zendeskgarden/react-tooltips';
+import { IconButton } from '@zendeskgarden/react-buttons';
+import { ReactComponent as MarkupStroke } from '@zendeskgarden/svg-icons/src/16/markup-stroke.svg';
+import { ReactComponent as CopyStroke } from '@zendeskgarden/svg-icons/src/16/copy-stroke.svg';
+import { ReactComponent as LightningBoltStroke } from '@zendeskgarden/svg-icons/src/16/lightning-bolt-stroke.svg';
+import { ReactComponent as DirectionLtrStroke } from '@zendeskgarden/svg-icons/src/16/direction-ltr-stroke.svg';
+import { ReactComponent as DirectionRtlStroke } from '@zendeskgarden/svg-icons/src/16/direction-rtl-stroke.svg';
+
+import { SyntaxHighlighter } from './components/SyntaxHighlighter';
+import { retrieveCodesandboxParameters } from './utils/retrieveCodesandboxParameters';
+import { copyToClipboard } from './utils/copyToClipboard';
+
+const StyledExampleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: ${p => p.theme.space.lg};
+  direction: ${p => p.theme.rtl && 'rtl'};
+`;
+
+const StyledFooterButton = styled(IconButton).attrs({ isPill: false, focusInset: true })`
+  margin-left: ${p => p.theme.space.xxs};
+`;
+
+export const CodeExample: React.FC<{ code: string }> = ({ children, code }) => {
+  const [isRtl, setIsRtl] = useState(false);
+  const [isCodeVisible, setIsCodeVisible] = useState(false);
+  const focusVisibleRef = useRef(null);
+  const initialTooltipContent = 'Copy to clipboard';
+  const [tooltipContent, setTooltipContent] = useState(initialTooltipContent);
+
+  const exampleTheme = useMemo<DefaultTheme>(() => {
+    return { ...DEFAULT_THEME, rtl: isRtl };
+  }, [isRtl]);
+
+  const parameters = useMemo(() => {
+    return retrieveCodesandboxParameters(code);
+  }, [code]);
+
+  useEffect(() => {
+    let timeout: any;
```

Comment:
👀 maybe this could be `number` type since `setTimeout` returns a number id?

Reply:
I was actually able to use the `NodeJS.Timeout` type correctly in this environment 🎉 updated